cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/PreferencesHttpHandler.java

```diff
@@ -0,0 +1,251 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.gateway.handlers;
+
+import co.cask.cdap.app.store.Store;
+import co.cask.cdap.app.store.StoreFactory;
+import co.cask.cdap.common.conf.Constants;
+import co.cask.cdap.config.PreferencesWrapper;
+import co.cask.cdap.gateway.auth.Authenticator;
+import co.cask.cdap.gateway.handlers.util.AbstractAppFabricHttpHandler;
+import co.cask.cdap.proto.Id;
+import co.cask.cdap.proto.ProgramType;
+import co.cask.http.HttpResponder;
+import com.google.gson.JsonSyntaxException;
+import com.google.inject.Inject;
+import org.jboss.netty.handler.codec.http.HttpRequest;
+import org.jboss.netty.handler.codec.http.HttpResponseStatus;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.QueryParam;
+
+/**
+ * Program Preferences HTTP Handler.
+ */
+@Path(Constants.Gateway.API_VERSION_3 + "/configuration/preferences")
```

Comment:
We agreed that we will leave this design as such. I will create a separate PR to address this - most likely using empty string for namespace (for things that fall outside of a specific namespace scope - ie, properties stored at instance level and console settings which are stored per user).